### ISSUE TITLE
Fix the http interceptor chain.

### ISSUE THREAD

**Issue by plabadie** [SAP/cloud-commerce-spartacus-storefront#1028]
### TL/DR

Let's remove the http interceptor providers from each individual modules and centralize them in the StorefrontModule.

### The problem

The http interceptor chain is broken.  What's happening is that we have http interceptors present multiple times in the interceptor chain.

<img width="1422" alt="wrong" src="https://user-images.githubusercontent.com/7065346/51755163-f9807880-208b-11e9-8973-8067fe34f217.png">

This kind of configuration issue is not obvious to diagnose and can result in very hard to troubleshoot issues.  In recent history, the client token refresh functionality was broken because of that.

The root cause of the issue was that the Auth module was imported twice, which has the side effect to provide the http interceptors defined in the Auth module twice.  This is most likely due to the very nature of multi-providers.

### The temporary fix

A quick fix of removing the AuthModule import from the CartModule will be merged with GH-470.  We can see that the interceptors are not duplicated anymore when we do that.

<img width="1365" alt="good" src="https://user-images.githubusercontent.com/7065346/51757439-57638f00-2091-11e9-9e5c-7989e303e023.png">

This is ok short term but, since a simple import is sufficient to break the interceptor chain, a more durable solution is required.

### The proposed solution

The [Angular HttpClient guide](https://angular.io/guide/http#provide-the-interceptor) suggests to use a centralized array in the app module to configure an application's http interceptors. ( as opposed to spread the interceptor providers in multiple module ).

> Consider creating a "barrel" file that gathers all the interceptor providers into an httpInterceptorProviders array

I see two advantages in centralizing the interceptor providers:
1) Avoid to accidentally create duplicate interceptors by importing a module more than once.
2) Gain control on the order each interceptor will have in the interceptor chain.  Interceptor declared in the same module keep their relative position within the interceptor chain.

*More precisely:*

While it doesn't make sense for us to declare the interceptors in the app module, the StorefrontModule is perhaps the next best place to provide the interceptors.  A downside to this is that the StorefrontModule is not in the same library as the interceptors themselves.  We could mitigate this by defining the httpInterceptorProviders array in the core lib and then import  the httpInterceptorProviders array from the StorefrontModule.  This way, http interceptor provider configuration will stay in the core lib.  It will only be read and applied in the StorefrontModule from the storefront lib.

**Comment by znikola** [comment]
To test:

1. Add console.log to i.e. auth interceptors' constructors.
2. Import `AuthModule` to `app.module.ts` like so:
```
  imports: [
     ...
    AuthModule
  ],
```
3. Monitor console log. You should see only one console log per interceptor. If you import the `AuthModule` by calling `AuthModule.forRoot()` you will see them twice.
4. You can repeat this for other interceptors
5. Test `MainModule` with special care, because it's a "collector" module that groups all UI-related modules.
It also groups two modules that have interceptors registered: `GlobalMessageComponentModule` and `HeaderModule`. Those two modules have `forRoot()` method defined in case a customer only wants to import them.
I tested by importing `MainModule` in `app.module.ts` and observing the browser's console (you should see one console log per interceptor). Then I imported `GlobalMessageComponentModule` next to `MainModule` and the result was the same: one console log per interceptor.

**Comment by LTiger14** [comment]
Passed QA
